### ISSUE TITLE
Reject connection when mac OS recover from screen lock.

### ISSUE THREAD

**Issue by peanut996** [SagerNet/sing-box#1709]
### Operating system

macOS

### System version

14.4.1

### Installation type

sing-box for macOS Graphical Client

### If you are using a graphical client, please provide the version of the client.

1.8.12

### Description

cannot continue proxy when recover from screen lock. 

<img width="747" alt="image" src="https://github.com/SagerNet/sing-box/assets/32303768/7095793f-9e26-406f-a60b-e55d7bc7d5bf">

### Reproduction

Just wait for screen lock.

### Logs

```shell
ERROR[3542][4215534725 0ms]inbound/mixed[mixed-in]: process connection from 127.0.0.1:56732: reject
msgstore.www.notion.so:443 while device paused
INFO[3542] [1418331084 0ms] inbound/mixed[mixed-in]: inbound connection from 127.0.0.1:56733
INFO[3542] [1418331084 0ms] inbound/mixed[mixed-in]: inbound connection to msgstore.www.notion.so:443
ERROR[3542] [1418331084 8ms] inbound/mixed[mixed-in]: process connection from 127.0.0.1:56733: reject connection tomsgstore.www.notion.so:443 while device paused
```

### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/nekohasekai/)

### Integrity requirements

- [X] I confirm that I have read the documentation, understand the meaning of all the configuration items I wrote, and did not pile up seemingly useful options or default values.
- [X] I confirm that I have provided the server and client configuration files and process that can be reproduced locally, instead of a complicated client configuration file that has been stripped of sensitive data.
- [X] I confirm that I have provided the simplest configuration that can be used to reproduce the error I reported, instead of depending on remote servers, TUN, graphical interface clients, or other closed-source software.
- [X] I confirm that I have provided the complete configuration files and logs, rather than just providing parts I think are useful out of confidence in my own intelligence.

**Comment by xionglingfeng** [comment]
I observed the same issue on my macOS as well. I believe its the same issue here:
https://github.com/SagerNet/sing-box/issues/1432
The bug affects both macOS and iOS.

**Comment by atc891** [comment]
confirm, i have the same problem on ios. I fixed this problem on 1.8.14, but don’t remember what i changed in config. After updating to 1.9.0 the problem returned
